File: src/main/java/ru/timmson/kanban/game/model/DiceSide.java
package ru.timmson.kanban.game.model;

import lombok.Builder;
import lombok.ToString;

import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@ToString
public class DiceSide {

    private Map<Stage, Integer> points;

    @Builder
    DiceSide(List<Work> works) {
        this.points = works.stream().collect(Collectors.toMap(Work::getStage, Work::getPoints));
    }

    public Integer getPoint(Stage stage) {
        return points.get(stage);
    }

}

File: src/main/kotlin/ru/timmson/kanban/game/model/DiceSide.kt
package ru.timmson.kanban.game.model

import com.fasterxml.jackson.annotation.JsonIgnore

data class DiceSide(@JsonIgnore var works: List<Work>) {

    private var points = mapOf<Stage?, Int?>()

    init {
        points = works.map { it.stage to it.points }.toMap()
    }

    fun getPoint(stage: Stage): Int? {
        return points[stage]
    }
}